### DISCUSSION TITLE
DevTools Protocol

### DISCUSSION THREAD

**Discussion by rlaffers** [statelyai/rfcs#15] [General]
Hi @davidkpiano ,
I'm developing an xstate devtools extension and would love to make it conform to your proposed interface. Regarding [your proposal](https://github.com/statelyai/rfcs/blob/davidkpiano/devtools-protocol/text/0000-devtools-protocol.md):

Shouldn't the register/unregister method signature accept **Interpreter** instead of Actor?
```
register: (actorRef: AnyInterpreter) => void;
```
Elsewhere the **ActorRegistration** type refers to sessionId prop, but AFAIK this exists only on state machine actors, a.k.a. Interpreters, and not on generic ActorRefs. Or am I missing something obvious? Do you envision that a sessionId should be generated by the register method if the actor does not have it?

**Comment by Andarist** [comment]
@rlaffers what kind of an extension are you developing?

**Reply by rlaffers** [reply]
Chrome/Firefox web extension to replace Redux devtools for XState debugging.
Eventually making an interface for NodeJS via the Devtools protocol.
